src/java/com/puppetlabs/puppetserver/JRubyPuppet.java

```diff
@@ -24,7 +24,7 @@
     List getTransportInfoForEnvironment(String environment);
     List getModuleInfoForEnvironment(String environment);
     Map compileCatalog(Map requestBody);
-    Map compileAST(Map compileOptions);
+    Map compileAST(Map compileOptions, List boltlibPath);
```

Comment:
I keep getting the following error. i dont understand why its `compile_ast`. What am i misssing?
```
689950503372607515.edn
{:clojure.main/message
 "Syntax error (IllegalArgumentException) compiling reify* at (puppetlabs/services/jruby/jruby_puppet_service.clj:17:1).\nCan't define method not in interfaces: compile_ast\n",
 :clojure.main/triage
 {:clojure.error/phase :compile-syntax-check,
  :clojure.error/line 17,
  :clojure.error/column 1,
  :clojure.error/source "jruby_puppet_service.clj",
  :clojure.error/symbol reify*,
  :clojure.error/path
  "puppetlabs/services/jruby/jruby_puppet_service.clj",
  :clojure.error/class java.lang.IllegalArgumentException,
  :clojure.error/cause
  "Can't define method not in interfaces: compile_ast"},
 :clojure.main/trace
 {:via
  [{:type clojure.lang.Compiler$CompilerException,
    :message
    "Syntax error compiling reify* at (puppetlabs/services/jruby/jruby_puppet_service.clj:17:1).",
    :data
    {:clojure.error/phase :compile-syntax-check,
     :clojure.error/line 17,
     :clojure.error/column 1,
     :clojure.error/source
     "puppetlabs/services/jruby/jruby_puppet_service.clj",
     :clojure.error/symbol reify*},
    :at [clojure.lang.Compiler analyzeSeq "Compiler.java" 7115]}
   {:type java.lang.IllegalArgumentException,
    :message "Can't define method not in interfaces: compile_ast",
    :at
    [clojure.lang.Compiler$NewInstanceMethod
     parse
     "Compiler.java"
     8529]}],
  :trace
  [[clojure.lang.Compiler$NewInstanceMethod parse "Compiler.java" 8529]
   [clojure.lang.Compiler$NewInstanceExpr build "Compiler.java" 8059]
   [clojure.lang.Compiler$NewInstanceExpr$ReifyParser
    parse
    "Compiler.java"
    7960]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler access$300 "Compiler.java" 38]
   [clojure.lang.Compiler$LetExpr$Parser parse "Compiler.java" 6384]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler$BodyExpr$Parser parse "Compiler.java" 6120]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler$BodyExpr$Parser parse "Compiler.java" 6120]
   [clojure.lang.Compiler$LetExpr$Parser parse "Compiler.java" 6436]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler$BodyExpr$Parser parse "Compiler.java" 6120]
   [clojure.lang.Compiler$LetExpr$Parser parse "Compiler.java" 6436]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler$BodyExpr$Parser parse "Compiler.java" 6120]
   [clojure.lang.Compiler$LetExpr$Parser parse "Compiler.java" 6436]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler$BodyExpr$Parser parse "Compiler.java" 6120]
   [clojure.lang.Compiler$LetExpr$Parser parse "Compiler.java" 6436]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler$BodyExpr$Parser parse "Compiler.java" 6120]
   [clojure.lang.Compiler$LetExpr$Parser parse "Compiler.java" 6436]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler$BodyExpr$Parser parse "Compiler.java" 6120]
   [clojure.lang.Compiler$LetExpr$Parser parse "Compiler.java" 6436]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler$BodyExpr$Parser parse "Compiler.java" 6120]
   [clojure.lang.Compiler$LetExpr$Parser parse "Compiler.java" 6436]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler$BodyExpr$Parser parse "Compiler.java" 6120]
   [clojure.lang.Compiler$FnMethod parse "Compiler.java" 5467]
   [clojure.lang.Compiler$FnExpr parse "Compiler.java" 4029]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7105]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler access$300 "Compiler.java" 38]
   [clojure.lang.Compiler$LetExpr$Parser parse "Compiler.java" 6384]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler$MapExpr parse "Compiler.java" 3104]
   [clojure.lang.Compiler analyze "Compiler.java" 6797]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler$BodyExpr$Parser parse "Compiler.java" 6120]
   [clojure.lang.Compiler$NewInstanceMethod parse "Compiler.java" 8553]
   [clojure.lang.Compiler$NewInstanceExpr build "Compiler.java" 8059]
   [clojure.lang.Compiler$NewInstanceExpr$ReifyParser
    parse
    "Compiler.java"
    7960]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7095]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler access$300 "Compiler.java" 38]
   [clojure.lang.Compiler$DefExpr$Parser parse "Compiler.java" 596]
   [clojure.lang.Compiler analyzeSeq "Compiler.java" 7107]
   [clojure.lang.Compiler analyze "Compiler.java" 6789]
   [clojure.lang.Compiler analyze "Compiler.java" 6745]
   [clojure.lang.Compiler eval "Compiler.java" 7181]
   [clojure.lang.Compiler load "Compiler.java" 7636]
   [clojure.lang.RT loadResourceScript "RT.java" 381]
   [clojure.lang.RT loadResourceScript "RT.java" 372]
   [clojure.lang.RT load "RT.java" 459]
   [clojure.lang.RT load "RT.java" 424]
   [clojure.core$load$fn__6839 invoke "core.clj" 6126]
   [clojure.core$load invokeStatic "core.clj" 6125]
   [clojure.core$load doInvoke "core.clj" 6109]
   [clojure.lang.RestFn invoke "RestFn.java" 408]
   [clojure.core$load_one invokeStatic "core.clj" 5908]
   [clojure.core$load_one invoke "core.clj" 5903]
   [clojure.core$load_lib$fn__6780 invoke "core.clj" 5948]
   [clojure.core$load_lib invokeStatic "core.clj" 5947]
   [clojure.core$load_lib doInvoke "core.clj" 5928]
   [clojure.lang.RestFn applyTo "RestFn.java" 142]
   [clojure.core$apply invokeStatic "core.clj" 667]
   [clojure.core$load_libs invokeStatic "core.clj" 5985]
   [clojure.core$load_libs doInvoke "core.clj" 5969]
   [clojure.lang.RestFn applyTo "RestFn.java" 137]
   [clojure.core$apply invokeStatic "core.clj" 667]
   [clojure.core$require invokeStatic "core.clj" 6007]
   [clojure.core$require doInvoke "core.clj" 6007]
   [clojure.lang.RestFn invoke "RestFn.java" 1289]
   [puppetlabs.enterprise.services.file_sync.file_sync_versioned_code_service$eval42445$loading__6721__auto____42446
    invoke
    "file_sync_versioned_code_service.clj"
    1]
   [puppetlabs.enterprise.services.file_sync.file_sync_versioned_code_service$eval42445
    invokeStatic
    "file_sync_versioned_code_service.clj"
    1]
   [puppetlabs.enterprise.services.file_sync.file_sync_versioned_code_service$eval42445
    invoke
    "file_sync_versioned_code_service.clj"
    1]
   [clojure.lang.Compiler eval "Compiler.java" 7177]
   [clojure.lang.Compiler eval "Compiler.java" 7166]
   [clojure.lang.Compiler load "Compiler.java" 7636]
   [clojure.lang.RT loadResourceScript "RT.java" 381]
   [clojure.lang.RT loadResourceScript "RT.java" 372]
   [clojure.lang.RT load "RT.java" 459]
   [clojure.lang.RT load "RT.java" 424]
   [clojure.core$load$fn__6839 invoke "core.clj" 6126]
   [clojure.core$load invokeStatic "core.clj" 6125]
   [clojure.core$load doInvoke "core.clj" 6109]
   [clojure.lang.RestFn invoke "RestFn.java" 408]
   [clojure.core$load_one invokeStatic "core.clj" 5908]
   [clojure.core$load_one invoke "core.clj" 5903]
   [clojure.core$load_lib$fn__6780 invoke "core.clj" 5948]
   [clojure.core$load_lib invokeStatic "core.clj" 5947]
   [clojure.core$load_lib doInvoke "core.clj" 5928]
   [clojure.lang.RestFn applyTo "RestFn.java" 142]
   [clojure.core$apply invokeStatic "core.clj" 667]
   [clojure.core$load_libs invokeStatic "core.clj" 5985]
   [clojure.core$load_libs doInvoke "core.clj" 5969]
   [clojure.lang.RestFn applyTo "RestFn.java" 137]
   [clojure.core$apply invokeStatic "core.clj" 667]
   [clojure.core$require invokeStatic "core.clj" 6007]
   [clojure.core$require doInvoke "core.clj" 6007]
   [clojure.lang.RestFn invoke "RestFn.java" 408]
   [puppetlabs.trapperkeeper.bootstrap$eval17780$resolve_service_BANG___17787$fn__17788$fn__17789
    invoke
    "bootstrap.clj"
    210]
   [puppetlabs.trapperkeeper.bootstrap$eval17780$resolve_service_BANG___17787$fn__17788
    invoke
    "bootstrap.clj"
    210]
   [puppetlabs.trapperkeeper.bootstrap$eval17780$resolve_service_BANG___17787
    invoke
    "bootstrap.clj"
    203]
   [puppetlabs.trapperkeeper.bootstrap$eval17834$resolve_and_handle_errors_BANG___17841$fn__17845
    invoke
    "bootstrap.clj"
    241]
   [puppetlabs.trapperkeeper.bootstrap$eval17834$resolve_and_handle_errors_BANG___17841
    invoke
    "bootstrap.clj"
    232]
   [clojure.core$map$fn__5866 invoke "core.clj" 2753]
   [clojure.lang.LazySeq sval "LazySeq.java" 42]
   [clojure.lang.LazySeq seq "LazySeq.java" 51]
   [clojure.lang.RT seq "RT.java" 535]
   [clojure.core$seq__5402 invokeStatic "core.clj" 137]
   [clojure.core$filter$fn__5893 invoke "core.clj" 2809]
   [clojure.lang.LazySeq sval "LazySeq.java" 42]
   [clojure.lang.LazySeq seq "LazySeq.java" 51]
   [clojure.lang.RT seq "RT.java" 535]
   [clojure.core$seq__5402 invokeStatic "core.clj" 137]
   [clojure.core$zipmap invokeStatic "core.clj" 3071]
   [clojure.core$zipmap invoke "core.clj" 3071]
   [puppetlabs.trapperkeeper.bootstrap$eval17754$check_duplicate_service_implementations_BANG___17761$fn__17762
    invoke
    "bootstrap.clj"
    196]
   [puppetlabs.trapperkeeper.bootstrap$eval17754$check_duplicate_service_implementations_BANG___17761
    invoke
    "bootstrap.clj"
    189]
   [puppetlabs.trapperkeeper.bootstrap$eval17920$parse_bootstrap_configs_BANG___17927$fn__17928
    invoke
    "bootstrap.clj"
    306]
   [puppetlabs.trapperkeeper.bootstrap$eval17920$parse_bootstrap_configs_BANG___17927
    invoke
    "bootstrap.clj"
    290]
   [puppetlabs.trapperkeeper.core$eval18216$boot_with_cli_data__18223$fn__18224
    invoke
    "core.clj"
    131]
   [puppetlabs.trapperkeeper.core$eval18216$boot_with_cli_data__18223
    invoke
    "core.clj"
    97]
   [puppetlabs.trapperkeeper.core$eval18245$run__18250$fn__18251
    invoke
    "core.clj"
    155]
   [puppetlabs.trapperkeeper.core$eval18245$run__18250
    invoke
    "core.clj"
    149]
   [puppetlabs.trapperkeeper.core$main invokeStatic "core.clj" 224]
   [puppetlabs.trapperkeeper.core$main doInvoke "core.clj" 210]
   [clojure.lang.RestFn applyTo "RestFn.java" 137]
   [clojure.lang.Var applyTo "Var.java" 705]
   [clojure.core$apply invokeStatic "core.clj" 665]
   [clojure.core$apply invoke "core.clj" 660]
   [puppetlabs.trapperkeeper.main$_main invokeStatic "main.clj" 7]
   [puppetlabs.trapperkeeper.main$_main doInvoke "main.clj" 4]
   [clojure.lang.RestFn applyTo "RestFn.java" 137]
   [clojure.lang.Var applyTo "Var.java" 705]
   [clojure.core$apply invokeStatic "core.clj" 665]
   [clojure.main$main_opt invokeStatic "main.clj" 514]
   [clojure.main$main_opt invoke "main.clj" 510]
   [clojure.main$main invokeStatic "main.clj" 664]
   [clojure.main$main doInvoke "main.clj" 616]
   [clojure.lang.RestFn applyTo "RestFn.java" 137]
   [clojure.lang.Var applyTo "Var.java" 705]
   [clojure.main main "main.java" 40]],
  :cause "Can't define method not in interfaces: compile_ast",
  :phase :compile-syntax-check}}

```